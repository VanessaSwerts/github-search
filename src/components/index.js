import CustomHeader from './Header/Header'
import CustomFooter from './Footer/Footer'
import Error from './Error/Error'
import Loading from './Loading/Loading'

export {
  CustomHeader,
  CustomFooter,
  Error,
  Loading
}