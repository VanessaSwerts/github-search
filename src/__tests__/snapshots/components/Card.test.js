import React from 'react'
import { render } from '@testing-library/react'
import { Card } from '../../../components'

describe('Testes de Snapshot da Card', () => {

  test('Renderização do componente Card', () => {
    const user = {
      name: "Example",
      username: "username",
      location: "location",
      followers: "0",
      following: "0",
      repos: "0",
      avatar: 'https://avatars.githubusercontent.com/u/57936?v=4'
    }

    const { asFragment } = render(<Card data={user} />)

    expect(asFragment(<Card />)).toMatchSnapshot()
  })

}) 