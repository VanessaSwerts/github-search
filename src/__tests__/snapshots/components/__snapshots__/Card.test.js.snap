// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes de Snapshot da Card Renderização do componente Card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-user"
      data-testid="card-user"
    >
      <div
        class="card-image"
      >
        <img
          alt="User Avatar"
          data-testid="card-avatar"
        />
      </div>
      <div
        class="card-content"
      >
        <h1
          class="card-name"
          data-testid="card-name"
        >
          Example
        </h1>
        <span
          class="card-username"
        >
          <i
            class="fas fa-user"
          />
          <span
            data-testid="card-username"
          />
        </span>
        <span
          class="card-location"
        >
          <i
            class="fas fa-map-marker-alt"
          />
          <span
            data-testid="card-location"
          >
            location
          </span>
        </span>
        <div
          class="card-data"
        >
          <span>
            <i
              class="fas fa-user-friends"
            />
            <span
              data-testid="card-following"
            >
              0
            </span>
            following
          </span>
          <span>
            <i
              class="fas fa-user-friends"
            />
            <span
              data-testid="card-followers"
            >
              0
            </span>
            followers
          </span>
          <span>
            <i
              class="fas fa-poll-h"
            />
            <span
              data-testid="card-repos"
            />
            repos
          </span>
        </div>
        <div
          class="card-footer"
        >
          <a
            href="https://github.com/undefined"
          >
            <span>
              GitHub 
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
